gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/GenerateWorkUnitsImpl.java

```diff
@@ -73,13 +77,22 @@ public int generateWorkUnits(Properties jobProps, EventSubmitterContext eventSub
       // before embarking on (potentially expensive) WU creation, first pre-check that the FS is available
       FileSystem fs = JobStateUtils.openFileSystem(jobState);
       fs.mkdirs(workDirRoot);
-
-      List<WorkUnit> workUnits = generateWorkUnitsForJobState(jobState, eventSubmitterContext, closer);
-
+      boolean canCleanUpTempDirs = false; // unlike `AbstractJobLauncher` running the job end-to-end, this is Work Discovery only, so WAY TOO SOON for cleanup
+      DestinationDatasetHandlerService datasetHandlerService = closer.register(
+          new DestinationDatasetHandlerService(jobState, canCleanUpTempDirs, eventSubmitterContext.create()));
+
+      List<WorkUnit> workUnits = generateWorkUnitsForJobState(jobState, datasetHandlerService, closer);
+
+      // GET FOLDERS FROM HERE TO CLEANUP
+      Set<String> foldersToCleanup = new HashSet<>();
+      for (WorkUnit workUnit : workUnits) {
+        foldersToCleanup.add(workUnit.getProp(ConfigurationKeys.WRITER_STAGING_DIR));
+        foldersToCleanup.add(workUnit.getProp(ConfigurationKeys.WRITER_OUTPUT_DIR));
+      }
```

Comment:
let's abstract within a named method.  that is more easily tested.  also, don't we need to check `getProp` for when that's not found?